### DISCUSSION TITLE
scan devices on the bus automatically

### DISCUSSION THREAD

**Discussion by monde38** [XKNX/xknx#971] [Q&A] [answered]
hi everyone, i'm using xknx and it's perfect for now. But i want to know if it's possible to connect with a real installation and automatically  
scan all the devices on the bus without having to create the object like:  
            **>>**  light1 = Light(knx, name='Switch (D7)', group_address_switch='1/1/0') 
basically i want to make a generic script for a building that can access all devices. 
thanks :)

**Comment by farmio** [comment, marked as answer]
Hi!

No, this is not possible.

One way would be to import the `.knxproj` file and parse it. There is already some work done by Marvin - see xknx/xknxproject repo. But even then I don't think full featured devices can be derived. Maybe if ETS functions are defined, but not sure how far this could take.

**Reply by monde38** [reply]
mmm... ok thank you! good luck!!

**Reply by monde38** [reply]
hi again,

I was trying to analyze a .knxproj as the example in the repo says but this error comes out, could someone help me please? I don't know which list the exception refers to
![image](https://user-images.githubusercontent.com/78907702/178109050-ee5793c8-9c24-4907-98d0-5af77cf42bf1.png)

**Reply by farmio** [reply]
I guess it was never tested on Windows. Maybe open an issue in the according repo reporting the bug en detail.

**Reply by monde38** [reply]
thanks you.

**Reply by monde38** [reply]
update:
after tracing the exception I could see that after line 29 the code flow goes directly to the line 35 and does not execute line 30, 31, 32, 33 and 34. I have no idea why. Perhaps for this reason it is that it is about accessing an index that does not exist and the exception is thrown. I'm using a .knxproj that i downloaded here -> https://www.theben.de/service-en-gb/ets-project-examples/ (all the examples work the same way) I appreciate any help.

![image](https://user-images.githubusercontent.com/78907702/178475916-82be79c6-96af-4713-8275-0aab6ac5b794.png)

**Reply by farmio** [reply]
Running it in a normal shell would show you a proper traceback. 
The Theben test-project has no building schema defined. xknxproject seems to expect one.
But this is the wrong repository to discuss this...
